app/components/show_more_button/show_more_button.js

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {Text, TouchableOpacity, View} from 'react-native';
+import LinearGradient from 'react-native-linear-gradient';
+
+import FormattedText from 'app/components/formatted_text';
+
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+export default class ShowMoreButton extends PureComponent {
+    static propTypes = {
+        highlight: PropTypes.bool,
+        onPress: PropTypes.func.isRequired,
+        showMore: PropTypes.bool.isRequired,
+        theme: PropTypes.object.isRequired,
+    };
+
+    static defaultProps = {
+        showMore: true,
+    };
+
+    renderButton(showMore, style) {
+        let sign = '+';
+        let textId = 'post_info.message.show_more';
+        let textMessage = 'Show More';
+        if (!showMore) {
+            sign = '-';
```

Comment:
Since we're using `+` on show more, I did make `-` on show less.  This is not mentioned in the ticket nor similar to webapp (where we used angle-up/down).  Let me know if this is good to go on UI perspective.